File: Raven.Client.Lightweight/FieldHighlightings.cs
using System.Collections.Generic;
using Raven.Abstractions.Data;

namespace Raven.Client
{
	/// <summary>
	///     Query highlightings for the documents.
	/// </summary>
	public class FieldHighlightings
	{
		private readonly Dictionary<string,string[]> highlightings;

		public FieldHighlightings(string fieldName)
		{
			this.FieldName = fieldName;
			this.highlightings = new Dictionary<string, string[]>();
		}

		/// <summary>
		///     The field name.
		/// </summary>
		public string FieldName { get; private set; }

		public IEnumerable<string> ResultIndents
		{
			get { return this.highlightings.Keys; }
		}

		/// <summary>
		///     Returns the list of document's field highlighting fragments.
		/// </summary>
		/// <param name="documentId">The document id.</param>
		/// <returns></returns>
		public string[] GetFragments(string documentId)
		{
			string[] result;

			if (!this.highlightings.TryGetValue(documentId, out result))
				return new string[0];

			return result;
		}

		internal void Update(QueryResult queryResult)
		{
			this.highlightings.Clear();

			foreach (var entityFragments in queryResult.Highlightings)
				foreach (var fieldFragments in entityFragments.Value)
					if (fieldFragments.Key == this.FieldName)
						this.highlightings.Add(entityFragments.Key, fieldFragments.Value);
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/FieldHighlightings.java
package net.ravendb.client;

import java.util.HashMap;
import java.util.Map;

import net.ravendb.abstractions.data.QueryResult;


/**
 *  Query highlightings for the documents.
 */
public class FieldHighlightings {
  private final Map<String, String[]> highlightings;
  private String fieldName;

  public FieldHighlightings(String fieldName) {
    this.fieldName = fieldName;
    this.highlightings = new HashMap<>();
  }

  public String getFieldName() {
    return fieldName;
  }


  public Map<String, String[]> getHighlightings() {
    return highlightings;
  }

  /**
   * Returns the list of document's field highlighting fragments.
   * @param documentId
   * @return
   */
  public String[] getFragments(String documentId) {
    if (this.highlightings.containsKey(documentId)) {
      return highlightings.get(documentId);
    } else {
      return new String[0];
    }
  }

  protected void update(QueryResult queryResult) {
    this.highlightings.clear();

    for (Map.Entry<String, Map<String, String[]>> entityFragments : queryResult.getHighlightings().entrySet()) {
      for(Map.Entry<String, String[]> fieldFragments : entityFragments.getValue().entrySet()) {
        if (fieldFragments.getKey().equals(this.fieldName)) {
          this.highlightings.put(entityFragments.getKey(), fieldFragments.getValue());
        }
      }
    }
  }

}
